Sources/CoreBusiness/DataProvider/Server.swift

```diff
@@ -7,29 +7,102 @@
 import Foundation
 
 /// A server environment.
-public enum Server {
+public struct Server {
+#if os(iOS)
+    private static let vector = "appplay"
+#else
+    private static let vector = "tvplay"
+#endif
+
     /// Production.
-    case production
+    public static let production = custom(baseUrl: URL(string: "https://il.srgssr.ch")!)
 
     /// Stage.
-    case stage
+    public static let stage = custom(baseUrl: URL(string: "https://il-stage.srgssr.ch")!)
 
     /// Test.
-    case test
-
-    /// MMF.
-    case mmf
-
-    var url: URL {
-        switch self {
-        case .production:
-            return URL(string: "https://il.srgssr.ch")!
-        case .stage:
-            return URL(string: "https://il-stage.srgssr.ch")!
-        case .test:
-            return URL(string: "https://il-test.srgssr.ch")!
-        case .mmf:
-            return URL(string: "https://play-mmf.herokuapp.com")!
+    public static let test = custom(baseUrl: URL(string: "https://il-test.srgssr.ch")!)
+
+    private let requestBuilder: (String) -> URLRequest
+    private let resizedImageUrlBuilder: (URL, ImageWidth) -> URL
```

Comment:
These two closures seem to be used many times, so my suggestion is to use type aliases.

```swift
public typealias RequestBuilder = (String) -> URLRequest
public typealias UrlBuilder = (URL, ImageWidth) -> URL
```

Reply:
I removed APIs for free construction with closures. Was overkill.